### DISCUSSION TITLE
Sharing memory pools between apps

### DISCUSSION THREAD

**Discussion by jhnphm** [nasa/cFS#368] [Q&A]
Is it a valid usecase for a CFE memory pool to be created in one app, then its handle transferred over to a global variable in a lib, and it to be manipulated in another app (indirectly)?

**Comment by jhnphm** [comment]
Hm, it seems through digging through SB that it uses the memory pool in exactly this fashion, so it might be okay...

**Reply by ghost** [reply]
It should be possible.. I don't know of a case outside of the cFE Core that does this. Does the ES memory pool API track the owner of a pool?

It's better to use messages when possible to keep apps decoupled, but sometimes this has to be done for performance.
